server/controller/apis/discussionController.js

```diff
@@ -0,0 +1,378 @@
+        res.set( "x-agora-message-title", "Bad Request" );
+        res.set( "x-agora-message-detail", "parent_id not provided" );
+        return res.status( 401 ).json( message );
+    }
+
+    if(req.body.parent_type !== "goal" && req.body.parent_type !== "topic" /*&& req.body.parent_type !== "comment"*/) {
+        const message = ApiMessage.createApiMessage( 401, "Bad Request", "parent_type not provided" );
+        res.set( "x-agora-message-title", "Bad Request" );
+        res.set( "x-agora-message-detail", "parent_type not provided" );
+        return res.status( 401 ).json( message );
+    }
+
+    if(req.body.comment_text === undefined || req.body.comment_text === null || req.body.comment_text === "") {
+        const message = ApiMessage.createApiMessage( 401, "Bad Request", "comment_text not provided" );
+        res.set( "x-agora-message-title", "Bad Request" );
+        res.set( "x-agora-message-detail", "comment_text not provided" );
+        return res.status( 401 ).json( message );
+    }
+
+    let comment = await discussionService.createComment(authUserId, req.body );
+
+    // not sure if this is possible, so I don't really know what the message should be
+    if(!comment) {
+        const message = ApiMessage.createApiMessage( 401, "Bad Request", "Comment could not be created" );
+        res.set( "x-agora-message-title", "Bad Request" );
+        res.set( "x-agora-message-detail", "Comment could not be created" );
+        return res.status( 401 ).json( message );
+    }
+    
+    res.set( "x-agora-message-title", "Success" );
+    res.set( "x-agora-message-detail", "Created comment" );
+    res.status( 200 ).json( comment );
+}
+
+exports.editComment = async ( req, res ) => {
+
+    const commentId = req.params.commentId
+    // get the user id either from the request user from basic auth in API call, or from the session for the UI
+    let authUserId;
+    if( req.user ) {
+        authUserId = req.user.id;
+    }
+    else if( req.session.authUser ) {
+        authUserId = req.session.authUser.id;
+    }
+
+    if(req.body.comment_text === undefined || req.body.comment_text === null || req.body.comment_text === "") {
+        const message = ApiMessage.createApiMessage( 401, "Bad Request", "comment_text not provided" );
+        res.set( "x-agora-message-title", "Bad Request" );
+        res.set( "x-agora-message-detail", "comment_text not provided" );
+        return res.status( 401 ).json( message );
+    }
+
+    let comment = await discussionService.editComment( commentId , authUserId , req.body );
+
+    if(!comment) {
+        const message = ApiMessage.createApiMessage( 404, "Not Found", "Comment not found" );
+        res.set( "x-agora-message-title", "Not Found" );
+        res.set( "x-agora-message-detail", "Comment not found" );
+        return res.status( 404 ).json( message );
+    }
+    
+    res.set( "x-agora-message-title", "Success" );
+    res.set( "x-agora-message-detail", "Returned all goals" );
```

Comment:
API message needs to be updated.
```suggestion
    res.set( "x-agora-message-detail", "Edited comment" );
```